### PR TITLE
Remove device maps from TensorPipe for v1.7 release

### PR DESCRIPTION
**Pull request by mrshenli**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#45353 Remove device maps from TensorPipe for v1.7 release**

Temporarily removing this feature, will add this back after branch cut.

Differential Revision: [D23939865](https://our.internmc.facebook.com/intern/diff/D23939865)

### CODE REVIEW

File: torch/csrc/distributed/rpc/tensorpipe_agent.cpp [hunk 1]
```diff
@@ -480,41 +477,16 @@ void TensorPipeAgent::sendCompletedResponseMessage(
...
-                  "TensorPipe RPC backend only supports CPU tensors by default."
-                  " Response device mapping is not available for destination ",
-                  pipe->getRemoteName(),
-                  " for device ",
-                  t.device(),
-                  " but received a tensor on that device."),
-              responseMessage.id());
-          break;
-        }
+    for (const auto& tensor : responseMessage.tensors()) {
```
Comment (rohan-varma) [inline torch/csrc/distributed/rpc/tensorpipe_agent.cpp:480]: What's the difference between this check here and `checkCPUTensor`? Could we use `checkCPUTensor` here also?

**Review by rohan-varma — APPROVED** [review]
I see that the error message we expect is "RPC backend only supports CPU tensors", but there is also an error string in `tensorpipe_agent.cpp`: "TensorPipe RPC backend only supports CPU tensors, please....". Which one do we prefer and when can the 2nd one get triggered?

File: torch/csrc/distributed/rpc/tensorpipe_agent.cpp (hunk 1 above)
Comment (rohan-varma) [reply torch/csrc/distributed/rpc/tensorpipe_agent.cpp:480]: (I mean the check right below this)

Comment (mrshenli) [reply torch/csrc/distributed/rpc/tensorpipe_agent.cpp:480]: `checkCPUTensor` is called in the user thread on sender, and hence should crash if it detects errors. This one is called on the callee, which should create an exception response instead.

**Comment by mrshenli** [comment]
> I see that the error message we expect is "RPC backend only supports CPU tensors", but there is also an error string in tensorpipe_agent.cpp: "TensorPipe RPC backend only supports CPU tensors, please....". Which one do we prefer and when can the 2nd one get triggered?

The `test_cuda` function will run for both ProcessGroupAgent and TensorPipeAgent. The 2nd message is triggered when using TensorPipe RPC backend, which adds a little more info to the error message.

**Pull request closed without merging**